packages/e2e-test-utils-playwright/src/admin/create-new-post.js

```diff
@@ -29,7 +31,11 @@ export async function createNewPost( {
 	} ).slice( 1 );
 
 	await this.visitAdminPage( 'post-new.php', query );
-	await this.page.waitForSelector( '.edit-post-layout' );
+	const canvasRoot = legacyCanvas
+		? this.page
+		: this.page.frameLocator( '[name=editor-canvas]' );
+
+	await canvasRoot.locator( '.editor-styles-wrapper' ).waitFor();
```

Comment:
Ideally we can. Though, this is on par with what it replaces `.edit-post-layout`. I'll take another look and see if I can find something.

Reply:
Maybe [`wp-embed-responsive`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/iframe/index.js#L152)? It's added on load for the iframe, but not for the legacy editor though.

Reply:
I've not been able to see anything that's clearly better. I'd started with the post title in my first experiments with this but had the thought that there could be a test for a custom post type that doesn't support a post title. I thought about the 'Add default block' button that's usually present but post types can have templates that won't include it. Maybe we don't and will never have tests for custom post types where this would be a problem.

Here's an idea, instead of the new `legacyCanvas: boolean` option I added to `createNewPost` it could be `waitFor: string`,  whose default is `'heading="Add title"'`. If we do have any tests that are for post types without titles they’ll have to be updated to pass in something else there. I'm happy to try that but wasn't sure it's any better. WDYT?

Reply:
The post title field and default block appender depend too much on post type settings. The title can be disabled and post-type content locked.

What about using the method from `visitSiteEditor`?

```js
if ( legacyCanvas ) {
	await this.page.waitForSelector( '.edit-post-layout' );
} else {
	// Wait for the iframed canvas contents to be rendered.
	await this.page
		.frameLocator( 'iframe[title="Editor canvas"i]' )
		.locator( 'body > *' )
		.first()
		.waitFor();
}
```

Reply:
I think that's better. Thanks George! See f585b1f. Style aside the main difference is avoiding `waitForSelector` because `ElementHandle` isn't encouraged in the docs.